pkgs/development/python-modules/wandb/default.nix

```diff
@@ -1,322 +1,324 @@
-      "tests/pytest_tests/unit_tests_old/tests_launch/test_launch_jobs.py"
-
-      # Requires google-cloud-aiplatform which is not packaged as of 2023-04-25.
-      "tests/pytest_tests/unit_tests/test_launch/test_runner/test_vertex.py"
-
-      # Requires google-cloud-artifact-registry which is not packaged as of 2023-04-25.
-      "tests/pytest_tests/unit_tests/test_launch/test_registry/test_gcp_artifact_registry.py"
-
-      # Requires kfp which is not packaged as of 2023-04-25.
-      "tests/pytest_tests/system_tests/test_core/test_kfp.py"
-
-      # Requires kubernetes_asyncio which is not packaged as of 2024-01-14.
-      "tests/pytest_tests/unit_tests/test_launch/test_builder/test_kaniko.py"
-      "tests/pytest_tests/unit_tests/test_launch/test_runner/test_kubernetes.py"
-      "tests/pytest_tests/unit_tests/test_launch/test_runner/test_safe_watch.py"
-
-      # Requires metaflow which is not packaged as of 2023-04-25.
-      "tests/pytest_tests/unit_tests/test_metaflow.py"
-
-      # Requires tensorflow which is broken as of 2023-09-03
-      "tests/pytest_tests/unit_tests/test_keras.py"
-
-      # Try to get hardware information, not possible in the nix build environment
-      "tests/pytest_tests/unit_tests/test_system_metrics/test_disk.py"
-
-      # See https://github.com/wandb/wandb/issues/5423
-      "tests/pytest_tests/unit_tests/test_docker.py"
-      "tests/pytest_tests/unit_tests/test_library_public.py"
-
-      # See https://github.com/wandb/wandb/issues/6836
-      "tests/pytest_tests/unit_tests_old/test_logging.py"
-    ]
-    ++ lib.optionals stdenv.hostPlatform.isLinux [
-      # Same as above
-      "tests/pytest_tests/unit_tests/test_artifacts/test_storage.py"
-    ]
-    ++ lib.optionals stdenv.hostPlatform.isDarwin [
-      # Same as above
-      "tests/pytest_tests/unit_tests/test_lib/test_filesystem.py"
-    ];
-
   disabledTests =
     [
-      # Timing sensitive
-      "test_login_timeout"
-
-      # Tensorflow 2.13 is too old for the current version of keras
-      # ModuleNotFoundError: No module named 'keras.api._v2'
-      "test_saved_model_keras"
-      "test_sklearn_saved_model"
-      "test_pytorch_saved_model"
-      "test_tensorflow_keras_saved_model"
+      # Probably failing because of lack of internet access
+      # AttributeError: module 'wandb.sdk.launch.registry' has no attribute 'azure_container_registry'. Did you mean: 'elastic_container_registry'?
+      "test_registry_from_uri"
+
+      # Require docker
+      "test_get_requirements_section_pyproject"
+      "test_local_custom_env"
+      "test_local_custom_port"
+      "test_local_default"
+
+      # Expects python binary to be named `python3` but nix provides `python3.12`
+      # AssertionError: assert ['python3.12', 'main.py'] == ['python3', 'main.py']
```

Comment:
```
❯ ns -p python312
[nix-shell:~]$ which python3
/nix/store/ybnf7k6i9p244bbhsbxizqk65z58cwyr-python3-3.12.6/bin/python3
```
perhaps I'm missing something here?

Reply:
I know, this is weird. But I confirm that the test fails:
```
_______________________ test_get_entrypoint[wandb_core] ________________________
[gw0] linux -- Python 3.12.6 /nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/bin/python3.12

    @pytest.mark.skipif(
        platform.system() == "Windows",
        reason="python exec name is different on windows",
    )
    def test_get_entrypoint():
        dir = tempfile.TemporaryDirectory().name
        job_source = "artifact"
        builder = _configure_job_builder_for_partial(dir, job_source)
    
        metadata = {"python": "3.9.11", "codePathLocal": "main.py", "_partial": "v0"}
    
        program_relpath = builder._get_program_relpath(job_source, metadata)
        entrypoint = builder._get_entrypoint(program_relpath, metadata)
>       assert entrypoint == ["python3", "main.py"]
E       AssertionError: assert ['python3.12', 'main.py'] == ['python3', 'main.py']
E         
E         At index 0 diff: 'python3.12' != 'python3'
E         
E         Full diff:
E           [
E         -     'python3',
E         +     'python3.12',
E         ?             +++
E               'main.py',
E           ]

/build/source/tests/unit_tests/test_launch/test_create_job.py:126: AssertionError
```